tests/legacy-cli/e2e/utils/project.ts

```diff
@@ -182,42 +187,96 @@ export function useCIDefaults(projectName = 'test-project'): Promise<void> {
   });
 }
 
+const KARMA_CONF_DEFAULT = `
```

Comment:
Ideally we don't create this file as this would cause the the default CLI `ng test` code path not to be covered by the tests at all.

Maybe add the ChromeHeadlessNoSandbox as a custom browser in the builder?

Reply:
Any examples for doing it in the builder like that?

I agree ideally this would not be here since it is not the standard user workflow that we want to be testing.

Reply:
It can be added here the custom browser in:
 https://github.com/angular/angular-cli/blob/df9674bd361b46b1275eed98e503f8897e701af5/packages/angular_devkit/build_angular/src/builders/karma/index.ts#L198

You can than use that browser by using the browsers option which can be set either in the CLI or angular.json 
```
ng test --browsers=ChromeHeadlessNoSandbox
```

Reply:
I got this working by adding the `ChromeHeadlessNoSandbox` to the karma builder. Is that what you meant? Adding it to the actual shipped code...?

Reply:
Yes.